samples/bluetooth/unicast_audio_client/src/main.c

```diff
@@ -38,39 +38,151 @@ static K_SEM_DEFINE(sem_stream_qos, 0, 1);
 static K_SEM_DEFINE(sem_stream_enabled, 0, 1);
 static K_SEM_DEFINE(sem_stream_started, 0, 1);
 
-void print_hex(const uint8_t *ptr, size_t len)
+
+#if defined(CONFIG_LIBLC3CODEC)
+
+#include "lc3_config.h"
+#include "lc3_encoder.h"
+#include "math.h"
+
+#define AUDIO_SAMPLE_RATE_HZ    16000
+#define AUDIO_FREQUENCY_HZ      800
+#define AUDIO_LENGTH_US         10000
+#define NUM_SAMPLES             ((AUDIO_LENGTH_US * AUDIO_SAMPLE_RATE_HZ) / 1000000)
+
+static int16_t audio_buf[NUM_SAMPLES];
+static Lc3Encoder_t lc3_encoder;
+
+/**
+ * Use the math lib to generate a sine-wave using 16 bit samples into a buffer.
+ *
+ * @param buf Destination buffer
+ * @param length_us Length of the buffer in microseconds
+ * @param frequency_hz frequency in Hz
+ * @param sample_rate_hz sample-rate in Hz.
+ */
+static void fill_audio_buf_sin(int16_t *buf, int length_us, int frequency_hz, int sample_rate_hz)
 {
-	while (len-- != 0) {
-		printk("%02x", *ptr++);
+	const int sine_period = sample_rate_hz / frequency_hz;
+	const int num_samples = (length_us * sample_rate_hz) / 1000000;
```

Comment:
num_samples is calculated rather than using NUM_SAMPLES as the intension is to use the selected codec configuration when lib functions are added to extract the data from the codec config.